RUFAS/e2e_test_results_comparer.py

```diff
@@ -0,0 +1,95 @@
+from collections import namedtuple
+from deepdiff import DeepDiff
+from pathlib import Path
+from typing import Any
+
+from RUFAS.input_manager import InputManager
+from RUFAS.output_manager import OutputManager
+from RUFAS.units import MeasurementUnits
+
+import json
+import re
+
+
+ResultPathType = namedtuple("ResultPaths", ["domain", "expected_results_path", "actual_results_path"])
+
+
+class E2ETestResultsComparer:
+    """
+    Contains routines to handle comparing actual and expected results for end-to-end testing of various RuFaS modules.
+    """
+
+    @staticmethod
+    def compare_actual_and_expected_test_results(json_output_path: Path) -> None:
+        """
+        Orchestrates the comparison between the expected and actual end-to-end testing results.
+
+        Parameters
+        ----------
+        json_output_directory : Path
+            Path to which JSON outputs are written to.
+
+        """
+        om = OutputManager()
+        info_map: dict[str, Any] = {
+            "class": E2ETestResultsComparer.__class__.__name__,
+            "function": E2ETestResultsComparer.compare_actual_and_expected_test_results.__name__,
+        }
+        test_result_path_sets = E2ETestResultsComparer._get_test_result_paths()
+
+        for path_set in test_result_path_sets:
+            om.add_log(
+                f"End-to-end testing for {path_set.domain}",
+                "Collecting and comparing actual and expected results",
+                info_map,
+            )
+            path_to_actual_results = None
+            for path in json_output_path.iterdir():
+                is_a_match = re.match(f"{str(json_output_path)}/{path_set.actual_results_path}.*", str(path))
+                if is_a_match:
+                    path_to_actual_results = path
+                    break
+            else:
+                om.add_error(
+                    f"Could not find actual end-to-end testing results for {path_set.domain}",
+                    "End-to-end testing failed.",
+                    info_map,
+                )
```

Comment:
```suggestion
                om.add_error(
                    "End-to-end testing failed.",
                    f"Could not find actual end-to-end testing results for {path_set.domain}",
                    info_map,
                )
```
Also, add 'path_set.domain` to `info_map`